backend/src/controller/chatController.ts

```diff
@@ -21,27 +23,82 @@ import { Defence } from '@src/models/defence';
 import { EmailInfo } from '@src/models/email';
 import { LEVEL_NAMES } from '@src/models/level';
 import { chatGptSendMessage } from '@src/openai';
+import { pushMessageToHistory } from '@src/utils/chat';
 
 import { handleChatError } from './handleError';
 
+function combineChatDefenceReports(
+	reports: ChatDefenceReport[]
+): ChatDefenceReport {
+	const combinedReport: ChatDefenceReport = {
+		blockedReason: reports
+			.filter((report) => report.blockedReason !== null)
+			.map((report) => report.blockedReason)
+			.join('\n'),
+		isBlocked: reports.some((report) => report.isBlocked),
+		alertedDefences: reports.map((report) => report.alertedDefences).flat(),
+		triggeredDefences: reports.map((report) => report.triggeredDefences).flat(),
+	};
+	return combinedReport;
+}
+
+function getChatHistoryUserMessages(
```

Comment:
Used by both `handleLowLevelChat` and `handleHigherLevelChat` to add the user's messages (and maybe transformed message) to the chat history.